public/sw.js

```diff
@@ -0,0 +1,65 @@
+importScripts('/assets/sw-toolbox/sw-toolbox.js');
+
+function networkOnlyError(request, values, options) {
+  return toolbox.networkOnly(request, values, options).catch((err) => {
+    return toolbox.cacheOnly(new Request('/offline'));
+  });
+}
+
+function networkFirstError(request, values, options) {
+  return toolbox.networkFirst(request, values, options).catch((err) => {
+    return toolbox.cacheOnly(new Request('/offline'));
+  });
+}
+
+function networkFirstIndex(request, values, options) {
+  return toolbox.networkFirst(request, values, options).catch((err) => {
+    return toolbox.cacheOnly(new Request('/'));
+  });
+}
+
+// pre cache all urls that are needed for the main and offline page
+var preCache = [
```

Comment:
The only really hard to maintain part of this is the preCache but i don't know how to get all the routes of the pages in there automatically and haven't really worked with scala or play... 

The rest of the routes shouldn't needed to be touched because all post and get requests are getting send to the right router as long as there is no problem with caching the get requests (i.e. the request only gets data and doesn't change anything on the server), the reason for the list of special cases is that some urls that are GET request really should be POST requests (e.g. admin/seed) and some are downloads that are too large to be cached without problems. Also some of the other routes there shouldn't really be necessary so i deleted some of them in a commit.